drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,345 @@
+	if (!priv->i2c) {
+		LOG_ERR("Could not get pointer to %s device", config->bus_name);
+		return -EINVAL;
+	}
+
+	/* Read Status register */
+	max17262_reg_read(priv, STATUS, &tmp);
+
+	if (!(tmp & STATUS_POR)) {
+		/*
+		 * Status.POR bit is set to 1 when MAX17262 detects that
+		 * a software or hardware POR event has occurred and
+		 * therefore a custom configuration needs to be set...
+		 * If POR event did not happen (Status.POR == 0), skip
+		 * init and continue with measurements.
+		 */
+		LOG_DBG("No POR event detected - skip device configuration");
+		return 0;
+	}
+	LOG_DBG("POR detected, setting custom device configuration...");
+
+	/** STEP 1 */
+	max17262_reg_read(priv, FSTAT, &tmp);
+
+	/* Do not continue until FSTAT.DNR bit is cleared */
+	while (tmp & FSTAT_DNR) {
+		k_sleep(K_MSEC(10));
+		max17262_reg_read(priv, FSTAT, &tmp);
+	}
+
+	/** STEP 2 */
+	/* Store original HibCFG value */
+	max17262_reg_read(priv, HIBCFG, &hibcfg);
+
+	/* Exit Hibernate Mode step 1 */
+	max17262_reg_write(priv, SOFT_WAKEUP, 0x0090);
+	/* Exit Hibernate Mode step 2 */
+	max17262_reg_write(priv, HIBCFG, 0x0000);
+	/* Exit Hibernate Mode step 3 */
+	max17262_reg_write(priv, SOFT_WAKEUP, 0x0000);
+
+	/** STEP 2.1 --> OPTION 1 EZ Config (No INI file is needed) */
+	/* Write DesignCap */
+	max17262_reg_write(priv, DESIGN_CAP, config->design_cap);
+
+	/* Write IChgTerm */
+	max17262_reg_write(priv, ICHG_TERM, config->desired_charging_current);
+
+	/* Write VEmpty */
+	max17262_reg_write(priv, VEMPTY, ((config->empty_voltage / 10) << 7) |
+					  ((config->recovery_voltage / 40) & 0x7F));
+
+	/* Write ModelCFG */
+	if (config->charge_voltage > 4275)
+		max17262_reg_write(priv, MODELCFG, 0x8400);
+	else
+		max17262_reg_write(priv, MODELCFG, 0x8000);
+
+	/*
+	 * Read ModelCFG.Refresh (highest bit),
+	 * proceed to Step 3 when ModelCFG.Refresh == 0
+	 */
+	max17262_reg_read(priv, MODELCFG, &tmp);
+
+	/* Do not continue until ModelCFG.Refresh == 0 */
+	while (tmp & MODELCFG_REFRESH) {
+		k_sleep(K_MSEC(10));
+		max17262_reg_read(priv, MODELCFG, &tmp);
+	}
+
+	/* Restore Original HibCFG value */
+	max17262_reg_write(priv, HIBCFG, hibcfg);
+
+	/** STEP 3 */
+	/* Read Status register */
+	max17262_reg_read(priv, STATUS, &tmp);
+
+	/* Do not continue until Status.POR bit is cleared */
+	while (tmp & STATUS_POR) {
+		tmp &= STATUS_POR_CLR; /* Clear PowerOnReset bit */
+		max17262_reg_write(priv, STATUS, tmp);
+		k_sleep(K_MSEC(10));
+		max17262_reg_read(priv, STATUS, &tmp);
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api max17262_battery_driver_api = {
+	.sample_fetch = max17262_sample_fetch,
+	.channel_get = max17262_channel_get,
+};
+
+#define MAX17262_INIT(index)						\
+	static struct max17262_data max17262_driver_##index;		\
+									\
+	static const struct max17262_config max17262_config_##index = { \
+		.bus_name = DT_INST_BUS_LABEL(index),			\
+		.design_voltage = DT_INST_PROP(index, design_voltage),	\
```

Comment:
Add a `.bus_address = DT_INST_REG_ADDR(index),` or so here to store the device I2C address.

Reply:
Done.